### PR TITLE
REL-1985: Nag NGO to check mask files

### PR DESCRIPTION
**Pull request by swalker2m**
This PR would add a new cron job tasked with sending a nagging email to the NGO and contact scientist of a program which has an unchecked mask definition file at least one week after it was last modified.  The tricky part is to ensure that the email is sent once and only once even if the ODB is down when the email would have normally been sent.  To accomplish that goal, we keep track of the time that the nagging email was sent, assuming it has been sent.  This `lastEmailed` property is kept in the `AuxFile` metadata along with the record of whether the file has been checked and the file's description.

In addition I extracted reusable bits of an existing mailer, the `edu.gemini.dbTools.ephemeris.ReportMailer`, into a shared `edu.gemini.dbTools.mailer.Mailer` for use by the `MaskCheckCron`.  There are now three types of mailer: production, development, and test.  The production mailer sends real emails to all intended recipients.  The development mailer (which we referred to as "forTesting" before) just writes the email that would have been sent.  The "test" mailer is meant to be used during the testing phase.  It attempts to send emails but removes any non-Gemini addresses before doing so.  It also modifies the subject and body of the email to make it clear that it is a test email.

### CODE REVIEW

File: bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/mailer/Mailer.scala [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          case ia: InternetAddress => Option(ia.getAddress).exists(_.trim.endsWith("@gemini.edu"))
+          case _                   => false
+        }
+
+      def send(m: MimeMessage): IO[Unit] = {
+        write(m) *> {
+
+          val allRecipients = m.getRecipients(TO)
+
+          val content       =
```
Comment (cquiroz) [inline bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/mailer/Mailer.scala]: Is there any need not to hardcode the messages and read them from config instead?

File: bundle/edu.gemini.auxfile.workflow/src/main/java/edu/gemini/auxfile/server/file/MetaData.java [hunk 2]
```diff
@@ -17,50 +22,57 @@
 public class MetaData {
 
 	private static final Logger LOGGER = Logger.getLogger(MetaData.class.getName());
-	
-    private static final String PROP_ROOT = "meta";
-    private static final String PROP_DESCRIPTION = "description";
-	private static final String PROP_CHECKED = "checked";
-    
+
+    private static final String PROP_ROOT         = "meta";
```
[the remark this answers is not in the document]
Comment (swalker2m) [reply bundle/edu.gemini.auxfile.workflow/src/main/java/edu/gemini/auxfile/server/file/MetaData.java]: I think this is a tabs vs. spaces issue.  It looks fine in my editor.  We should do a big PR that just de-tabbifies all the source code.

File: bundle/edu.gemini.auxfile.workflow/src/main/java/edu/gemini/auxfile/server/file/MetaData.java [hunk 3]
```diff
@@ -102,24 +124,34 @@ private void store() throws IOException {
...
 
 			// Read the PIO XML document.
 			ParamSet node = (ParamSet) PioXmlUtil.read(xml);
-		
+
 			// Get the metadata properties
 			checked = Pio.getBooleanValue(node, PROP_CHECKED, false);
 			description = Pio.getValue(node, PROP_DESCRIPTION, null);
+			lastEmailed = ImOption.apply(Pio.getValue(node, PROP_LAST_EMAILED, null)).map(s ->
+	            TIME_FORMAT.parse(s, Instant::from)
```
[the remark this answers is not in the document]
Comment (swalker2m) [reply bundle/edu.gemini.auxfile.workflow/src/main/java/edu/gemini/auxfile/server/file/MetaData.java]: Here too it looks fine in the editor.  I'll strip the file of tabs.

File: bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/mailer/Mailer.scala (hunk 1 above)
Comment (swalker2m) [reply bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/mailer/Mailer.scala]: Here it is just wrapping whatever the actual message is in some text that explains that this is just a test.

The REL task didn't mention any specific text for the production version of the email itself so I invented something for a first pass.  I expect they'll want to modify that text, but that it won't change so much that we need to go to the trouble of putting it in a config file.

**Pull request merged**